VirtoCommerce.Platform.Web/Scripts/app/dynamicProperties/blades/propertyValue-list.js

```diff
@@ -9,47 +9,49 @@
 
     blade.refresh = function () {
         blade.data = blade.currentEntity;
-        var rawProperties = angular.copy(blade.currentEntity.dynamicProperties);
-
-        _.each(rawProperties, function (x) {
-            if (x.values) {
-                x.values.sort(function (a, b) {
-                    return a.value && b.value
-                        ? (a.value.name
-                            ? a.value.name.localeCompare(b.value.name)
-                            : angular.isString(a.value) && angular.isString(b.value)
-                                ? a.value.localeCompare(b.value)
-                                : a.value < b.value ? -1 : a.value > b.value ? 1 : 0)
-                        : -1;
-                });
-            }
-        });
+        dynamicPropertiesApi.getAll({typeName: blade.currentEntity.objectType}, function (response) {
+            var rawProperties = response;
+
+            _.each(rawProperties, function(property) {
+                property.values = [];
+                var filteredProperty = _.find(blade.currentEntity.dynamicProperties, function (item) { return item.id === property.id; });
+                if (filteredProperty) {
+                    property.values = filteredProperty.values;
+                }
+            })
+
+            _.each(rawProperties, function (x) {
```

Comment:
'x' - is bad for use, 'rawProperty' - better

Reply:
I agree! this is also legacy code